Dockerfile

```
@@ -491,45 +453,115 @@ ENV CGO_CFLAGS="${CFLAGS}"
 ENV CGO_CXXFLAGS="${CXXFLAGS}"
 ENV CGO_LDFLAGS="${LDFLAGS}"
 
+ENV GO111MODULE="auto"
+
+ENV AWS_LC_FIPS_VER="2.0.9"
+
+USER root
+RUN dnf -y install golang
+
+ENV GO121VER="1.21.9"
+ENV GO122VER="1.22.1"
+
+# =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=
+
+FROM sdk-go-prep as sdk-go-1.21-prep
+
+USER builder
+
+WORKDIR /home/builder/sdk-go-${GO121VER}
```

nit: can simplify and make this more like the `aws-lc` build:
```suggestion
WORKDIR /home/builder/sdk-go
```

(since the version is implicitly embedded in the build stage and controlled by the hash file we copy in)